apis/v1alpha2/validation/gateway.go

```diff
@@ -44,7 +46,32 @@ func validateGatewaySpec(spec *gatewayv1a2.GatewaySpec, path *field.Path) field.
 // to the Gateway API specification.
 func validateGatewayListeners(listeners []gatewayv1a2.Listener, path *field.Path) field.ErrorList {
 	// TODO [danehans]: Add additional validation of listener fields.
-	return validateListenerHostname(listeners, path)
+	var errs field.ErrorList
+	e1:=validateListenerTLSConfig(listeners, path)
+	errs = append(errs,e1...)
+	e2:=validateListenerHostname(listeners, path)
+	errs = append(errs,e2...)
+	return errs
+}
+
+func validateListenerTLSConfig(listeners []gatewayv1a2.Listener,path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	tlsHttpsProtocols := []gatewayv1a2.ProtocolType{gatewayv1a2.HTTPSProtocolType,gatewayv1a2.TLSProtocolType}
+	for i,l := range listeners{
+		if !contains(tlsHttpsProtocols,l.Protocol) && l.TLS != nil {
```

Comment:
Our protocol validation allows for custom protocols. That means we actually have to invert this logic to only prevent these fields from being set if the protocol is "TCP", "UDP", or "HTTP".